app/controllers/developer_controller.rb

```diff
@@ -13,7 +13,28 @@ def update_sites
 
   def production_log
     @log = if params[:grep].present?
-             `tail -n 10000 log/production.log | grep -E '#{params[:grep]}' -C #{params[:context]} --color=never`
+             grep_str = if params[:grep].instance_of?(String)
+                          params[:grep]
+                        elsif params[:grep].respond_to?(to_s)
+                          params[:grep].to_s
+                        else
+                          flash[:warning] = 'Coercion failure on params[:grep]'
+                          redirect_to(root_path) && return
+                        end
+             context_int = if params[:context].instance_of?(Integer)
+                             params[:context]
+                           elsif params[:context].respond_to?(to_i)
```

Comment:
Colon before `:to_i`

Reply:
I actually am very curious about how this didn't make CI fail with a NameError. What you're doing is equivalent to `params[:context].respond_to?(to_i())`, which is to say that you're passing the return value of the `to_i` method to `respond_to?`. Now, there's no reason why the `to_i` method should be defined within an instance of ApplicationController, the only reason I can imagine this not just throwing an error is because rails does some weird `method_missing` magic which I kind of want to dig in to, but I gotta run so maybe later.